src/libs/antares/study/parts/thermal/cluster_load.cpp

```diff
@@ -0,0 +1,247 @@
+/*
+** Copyright 2007-2023 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+#include "../../../inifile.h"
+#include "../../../logs.h"
+
+#include "cluster.h"
+#include "cluster_load.h"
+
+namespace Antares::Data
+{
+
+ThermalClusterReader::ThermalClusterReader()
+{
+    for (auto const& [key, val] : Pollutant::namesToEnum)
+    {
+        callbackMap.emplace(key, [val](ThermalCluster& c, const IniFile::Property& p) {
+            return c.emissions.emissionFactors[val] = p.value.to<double>();
+        });
+    }
+
+    callbackMap.emplace("annuityinvestment", [](ThermalCluster& c, const IniFile::Property& p)
+            { return c.annuityInvestment = p.value.to<uint>(); });
+}
+
+bool ThermalClusterReader::loadFromProperty(ThermalCluster& cluster, const IniFile::Property* p)
+{
+    if (p->key.empty())
+        return false;
+    std::function<bool(ThermalCluster&, const IniFile::Property&)> cb;
+    try {
+        const std::string k = p->key.c_str();
+        cb = callbackMap.at(k);
+    } catch(const std::out_of_range& oor) {
+        logs.error() << "Property not found: " << p->key;
+        return false;
+    }
+    return cb(cluster, *p);
+}
+
+
+bool ThermalClusterReader::legacyLoadFromProperty(ThermalCluster& cluster, const IniFile::Property* p)
```

Comment:
I think we'll keep the kind of ugly `if..else` statements. We'll investigate serialization later on, but we don't want to re-invent the wheel.

Please remove the map, place everything in the `legacyLoadFromProperty` function and rename this function.